libmamba/src/core/pool.cpp

```diff
@@ -160,6 +160,25 @@ namespace mamba
             return false;
         }
 
+        bool
+        subdir_match(std::string candidate_url, std::string needle_spec)
+        {
+            std::string needle_channel = split(needle_spec, ":", 1)[0];
+            if (!contains(needle_channel, "/")){
+                return true;
+            }
+            std::string needle_subdir = rsplit(needle_channel, "/", 1)[1];
+
+            std::string candidate_subdir = rsplit(candidate_url, "/", 1)[1];
+
+            if (candidate_subdir == needle_subdir){
+                return true;
+            }
+            throw std::runtime_error(
```

Comment:
Not sure if this is the right way to handle this case here

Reply:
I'm not sure myself, open for any suggestions!